src/alda/sound/midi.clj

```diff
@@ -118,6 +118,22 @@
                   channels (.getChannels ^Synthesizer synth)]]
       (load-instrument! patch (aget channels channel)))))
 
+(defn- load-instrument-receiver! [patch-number ^Integer channel-number receiver]
+  (let [instrumentMessage (doto (new ShortMessage)
+                            (.setMessage ShortMessage/PROGRAM_CHANGE
+                                         ;; TODO why does this need to be -1
+                                         channel-number (dec patch-number) 0))]
+    (.send receiver instrumentMessage 0)))
+
+(defn load-instruments-receiver!
```

Comment:
Similar to my comment about `schedule-wait!`, there is some repeated code here between `load-instruments-receiver!` and `load-instruments!`.

I think this would be cleaner if the new implementation were factored into the existing `load-instruments!` function, using `*use-midi-sequencer*` to decide which implementation to use internally.

Reply:
Like the previous example, this would require us to pass in a `receiver` object into the `load-instruments!` function. Do you know a good way to solve that? I'm not good enough with clojure to know if there is :P

Reply:
I don't think we can keep it at the top level (since we can't keep a receiver around, it's tied to a sequencer or other midi object), and putting it on the audio context seems like a bit of a hack. I disliked the duplication as well though, so I found a way to merge these functions together with clojure optional args, which I think was the cleanest solution. Let me know if you don't like it though!